test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -418,6 +418,17 @@ public static String randomPositiveTimeValue() {
     }
 
     /**
+     * helper to get a random value in a certain range that's different from the input
+     */
+    public static <T> T randomValueOtherThanMany(Collection<T> input, Supplier<T> randomSupplier) {
```

Comment:
Why not make it take a `randomValueOtherThan(Predicate<T>, Supplier<T>)`? Then you can implement the other `randomValueOtherThan` using `input::equals` or something like that!